fvm/src/machine/default.rs

```diff
@@ -236,29 +236,32 @@ where
     }
 
     fn transfer(&mut self, from: ActorID, to: ActorID, value: &TokenAmount) -> Result<()> {
-        if from == to {
+        if from == to || value.is_zero() {
             return Ok(());
         }
+
         if value.is_negative() {
-            return Err(syscall_error!(SysErrForbidden;
+            return Err(syscall_error!(IllegalArgument;
                 "attempted to transfer negative transfer value {}", value)
             .into());
         }
 
+        // If the from actor doesn't exist, we return "insufficient funds" to distinguish between
+        // that and the case where the _receiving_ actor doesn't exist.
         let mut from_actor = self
             .state_tree
             .get_actor_id(from)?
             .context("cannot transfer from non-existent sender")
-            .or_error(ExitCode::SysErrSenderInvalid)?;
+            .or_error(ErrorNumber::InsufficientFunds)?;
```

Comment:
Worst case, we hit this and exit with the correct exit code.

But I think we _can_ end up here because an actor can:

1. Delete itself.
2. Try to send funds.

That's why I'm trying to avoid fatal errors as much as possible.